src/jit/lowerxarch.cpp

```diff
@@ -2366,104 +2478,468 @@ void Lowering::ContainCheckHWIntrinsic(GenTreeHWIntrinsic* node)
     HWIntrinsicCategory category    = Compiler::categoryOfHWIntrinsic(intrinsicID);
     HWIntrinsicFlag     flags       = Compiler::flagsOfHWIntrinsic(intrinsicID);
     int                 numArgs     = Compiler::numArgsOfHWIntrinsic(node);
-    GenTree*            op1         = node->gtGetOp1();
-    GenTree*            op2         = node->gtGetOp2();
+    var_types           baseType    = node->gtSIMDBaseType;
+
+    GenTree* op1 = node->gtGetOp1();
+    GenTree* op2 = node->gtGetOp2();
+    GenTree* op3 = nullptr;
 
     if ((flags & HW_Flag_NoContainment) != 0)
```

Comment:
Since you're in the vicinity of this, I think this would be much clearer as:
```
bool supportsContainment = ((flags & HW_Flag_NoContainment) != 0);
if (!supportsContainment)
```
Of course, what I'd really like is to reverse the sense of this flag.

Reply:
I'm working on a cleanup here as I finish going through the remaining intrinsics for cleanup.